packages/devtools_app/integration_test/run_tests.dart

```diff
@@ -61,6 +61,13 @@ Future<void> _runTest(
   List<String> testRunnerArgs,
   String testFilePath,
 ) async {
+  final parsedArgs = TestRunnerArgs(testRunnerArgs);
+  final testAppDevice = TestAppDevice.fromArgName(parsedArgs.testAppDevice)!;
+  if (!testAppDevice.supportsTest(testFilePath)) {
+    // Skip test, since it is not supported for device.
+    return;
+  }
+
```

Comment:
```suggestion
  TestRunnerArgs testRunnerArgs
) async {
  if (!testRunnerArgs.testAppDevice.supportsTest(testRunnerArgs.testTarget)) {
    // Skip test, since it is not supported for device.
    return;
  }

```